issuer-list/src/main/java/com/adyen/checkout/issuerlist/IssuerListComponent.java

```diff
@@ -35,23 +38,45 @@
     @SuppressWarnings("LambdaLast")
     public IssuerListComponent(@NonNull GenericPaymentMethodDelegate genericPaymentMethodDelegate, @NonNull IssuerListConfiguration configuration) {
         super(genericPaymentMethodDelegate, configuration);
-        initIssuers(genericPaymentMethodDelegate.getPaymentMethod().getIssuers());
+        initIssuers(genericPaymentMethodDelegate.getPaymentMethod());
     }
 
     MutableLiveData<List<IssuerModel>> getIssuersLiveData() {
         return mIssuersLiveData;
     }
 
-    private void initIssuers(@Nullable List<Issuer> issuerList) {
-        if (issuerList != null) {
-            final List<IssuerModel> issuerModelList = new ArrayList<>();
-            for (Issuer issuer : issuerList) {
-                if (!issuer.isDisabled()) {
-                    final IssuerModel issuerModel = new IssuerModel(issuer.getId(), issuer.getName());
-                    issuerModelList.add(issuerModel);
+    private void initIssuers(@NonNull PaymentMethod paymentMethod) {
+        final List<Issuer> issuersList = paymentMethod.getIssuers();
+        if (issuersList != null) {
+            initIssuers(issuersList);
+        } else {
+            initLegacyIssuers(paymentMethod.getDetails());
+        }
+    }
+
+    private void initIssuers(@NonNull List<Issuer> issuerList) {
```

Comment:
Maybe rename one of the 2 methods? I think it's a bit confusing to have 2 `initIssuers` even though they take different parameters. Your choice.

Reply:
👍